app/handlers/callback.py

```diff
@@ -68,13 +68,17 @@ def execute_put(self, *args, **kwargs):
     def _post(self, *args, **kwargs):
         try:
             lab_name = self.get_query_argument(models.LAB_NAME_KEY)
+            status_string = self.get_argument(models.LAVA_STATUS_STR_KEY)
             req_token = kwargs["token"]
             valid_lab, error = self._is_valid_token(req_token, lab_name)
             if not valid_lab:
                 response = hresponse.HandlerResponse(403)
                 response.reason = (
                     "Provided authentication token is not associated with "
                     "lab '%s' or is not valid" % lab_name)
+            elif status_string == models.LAVA_CANCELED_STATUS_STR:
+                response = hresponse.HandlerResponse(400)
```

Comment:
@mgalka is there any case that you will check is the status is incomplete? Just asking, cause you created a static  variable(LAVA_INCOMPLETE_STATUS_STR) but seems that you don't use. So, when you have a incomplete status, can you carry incomplete data? does it makes any sense?

Reply:
It was added for the sake of consistency. Although it wasn't used it was put there to reflect all possible statues. Nevertheless, I'm handling the issue differently at the moment.